ankihub/main/review_data.py

```diff
@@ -91,3 +93,55 @@ def _get_first_and_last_review_datetime_for_ah_deck(
 
 def _ms_timestamp_to_datetime(timestamp: int) -> datetime:
     return datetime.fromtimestamp(timestamp / 1000)
+
+
+def get_daily_review_data_since_last_sync(
+    last_sync: datetime,
+) -> List[DailyCardReviewSummaryData]:
+    """Filter revlog entries by the date of the last sync and the 'yesterday' date,
+    group by days, and compile the data."""
+    yesterday = datetime.now() - timedelta(days=1)
+    timestamp_last_sync_ms = int(datetime.timestamp(last_sync) * 1000)
+    timestamp_yesterday_ms = int(datetime.timestamp(yesterday) * 1000)
+
+    rows = aqt.mw.col.db.all(
+        """
+        SELECT r.id, r.ease, r.time
+        FROM revlog as r
+        JOIN cards as c ON r.cid = c.id
+        WHERE r.id BETWEEN ? AND ?
+        """,
+        timestamp_last_sync_ms,
+        timestamp_yesterday_ms,
+    )
```

Comment:
:wrench: Shouldn't we use `end_of_yesterday` (as in the suggested code below) if we want to get the data for the whole last day? Otherwise the cut-off time is different depending on the current time.

```suggestion
    """Filter revlog entries between the date of the last sync and the end of yesterday
    group by days, and compile the data."""
    end_of_yesterday = datetime.now().replace(
        hour=0, minute=0, second=0, microsecond=0
    ) - timedelta(microseconds=1)
    timestamp_last_sync_ms = int(datetime.timestamp(last_sync) * 1000)
    timestamp_end_of_yesterday = int(datetime.timestamp(end_of_yesterday) * 1000)

    rows = aqt.mw.col.db.all(
        """
        SELECT r.id, r.ease, r.time
        FROM revlog as r
        JOIN cards as c ON r.cid = c.id
        WHERE r.id BETWEEN ? AND ?
        """,
        timestamp_last_sync_ms,
        timestamp_end_of_yesterday,
    )
```